### PR TITLE
Added configuration for View instance variable name in templates

### PR DESCRIPTION
**Pull request by othercorey**
This let's you override the default `_view` template variable name.

### CODE REVIEW

File: src/View/TwigView.php [hunk 1]
```diff
@@ -238,11 +238,12 @@ protected function initializeProfiler(): void
      */
     protected function _render(string $templateFile, array $data = []): string
     {
+        $viewName = Configure::read('TwigView.context.view', '_view');
```
Comment (ADmad) [inline src/View/TwigView.php:241]: Why not use a view option instead of global config?

Comment (othercorey) [reply src/View/TwigView.php:241]: You mean a property? I just forgot to use it.

Comment (ADmad) [reply src/View/TwigView.php:241]: I just implemented it https://github.com/cakephp/twig-view/pull/23 :slightly_smiling_face:

File: config/config.sample.php [hunk 2]
```diff
@@ -9,5 +9,9 @@
         'markdown' => [
             'engine' => 'engine', // See https://twig.symfony.com/doc/3.x/filters/markdown_to_html.html and then set to `new DefaultMarkdown()`
         ],
+        'context' => [
+            // Name of Twig template variable name that contains View instance
+            'view' => '_view',
```
Comment (ADmad) [inline config/config.sample.php:14]: The key name could be `viewVar`. Could we use default var name `this` instead of `_view`?

Comment (othercorey) [reply config/config.sample.php:14]: If we're changing the configuration, then backwards compatibility is gone. I'd be open to changing that default name then ... however, we're making it rough to transition anything large here.

Comment (ADmad) [reply config/config.sample.php:14]: We are making the var name is configurable so anyone can set it to old name if needed.

Comment (othercorey) [reply config/config.sample.php:14]: I know ... but will they realize why? I guess we can explain the configuration change at the top of the readme.

**Comment by ADmad** [comment]
BakeView will need to be updated to keep using old name.

File: src/Twig/Node/Element.php [hunk 3]
```diff
@@ -74,7 +74,7 @@ public function compile(Compiler $compiler)
     {
         $compiler->addDebugInfo($this);
 
-        $compiler->raw('echo $context[\'_view\']->element(');
+        $compiler->raw('echo $context[\'this\']->element(');
```
Comment (ADmad) [inline src/Twig/Node/Element.php:77]: The var name is hard-coded here. This means if someone changes the var name it will stop working isn't it?

Comment (othercorey) [reply src/Twig/Node/Element.php:77]: That is very true. I can extend the custom variable unit test to include element and cell. Hopefully we remember if adding another one.

**Comment by dereuromark** [comment]
I just checked
this means now

    {{ _view.fetch('content')|raw }}

etc are not working anymore in existing code right?

**Comment by ADmad** [comment]
@dereuromark No, that never worked, I was mistaken.

**Comment by dereuromark** [comment]
Ah ok, this was in the .md examples in the repo.
And for my tests it seemed to work actually :)

**Comment by ADmad** [comment]
Err sorry `_view.fetch()` should work

**Comment by dereuromark** [comment]
With these changes it now doesnt anymore. So careful with BC here, maybe we need to outline this.

**Comment by othercorey** [comment]
The readme need a re-write when we are done with all these changes. You can enable BC by setting the `viewVar` config to `_view`.

**Comment by ADmad** [comment]
I just realized that view variables are passed to view using view options key `viewVars` so using `viewVar` to customize View instance name is a bit too similar. It would be better to use `viewVarName`.

**Comment by othercorey** [comment]
That's probably still too similar. How about `contextName`.

**Comment by ADmad** [comment]
`contextName` sounds good.

**Pull request merged**